### PR TITLE
Make spring boot service name detector handle BOOT-INF/classes

### PR DESCRIPTION
**Pull request by laurit**
When spring boot application is packaged in one jar `application.properties` and `application.yml` are under `BOOT-INF/classes/`.

### CODE REVIEW

File: instrumentation/spring/spring-boot-resources/testing/src/test/java/io/opentelemetry/instrumentation/spring/resources/TestBootInfClassesResource.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  @Test
+  void testServiceName() {
+    // verify that the test app, that is added as a dependency to this project, has the expected
+    // layout
+    assertThat(getClass().getResource("/application.properties")).isNull();
+    assertThat(getClass().getResource("/BOOT-INF/classes/application.properties")).isNotNull();
+
+    SpringBootServiceNameDetector guesser = new SpringBootServiceNameDetector();
+    Resource result = guesser.createResource(config);
```
Comment (breedx-splk) [inline instrumentation/spring/spring-boot-resources/testing/src/test/java/io/opentelemetry/instrumentation/spring/resources/TestBootInfClassesResource.java:30]:
```suggestion
    SpringBootServiceNameDetector detector = new SpringBootServiceNameDetector();
    Resource result = detector.createResource(config);
```

**Review by breedx-splk — APPROVED** [review]
Looks good to me!

File: instrumentation/spring/spring-boot-resources/testing/src/test/java/io/opentelemetry/instrumentation/spring/resources/TestBootInfClassesResource.java (hunk 1 above)
Comment (laurit) [reply instrumentation/spring/spring-boot-resources/testing/src/test/java/io/opentelemetry/instrumentation/spring/resources/TestBootInfClassesResource.java:30]: This is copy pasted from
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/195fd38c2c5eafcf918f839ee51f68bc6f4b3d03/instrumentation/spring/spring-boot-resources/library/src/test/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetectorTest.java#L49-L50
WDYT about renaming this in both tests after this is merged?

**Review by trask — APPROVED** [review]

File: instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java [hunk 2]
```diff
@@ -265,6 +265,18 @@ private String loadFromClasspath(String filename, Function<InputStream, String>
 
   // Exists for testing
   static class SystemHelper {
+    private final ClassLoader classLoader;
+    private final boolean addBootInfPrefix;
+
+    SystemHelper() {
+      ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
```
[the remark this answers is not in the document]
Comment (laurit) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:272]: There isn't anything that says this has to be used only with javaagent. I think the context class loader defaults to system when it is not set.

**Pull request merged**